### ISSUE TITLE
Integration with docker health check not working, Service fabric does not show unhealthy events in explorer and container never restarts if unhealthy.

### ISSUE THREAD

**Issue by adityapkanthe** [Azure/service-fabric-issues#922]
I am trying to integrate docker healthcheck with service fabric, I want service fabric to restart the container when it turns unhealthy, but this is not working for me. The container was unhealthy for more than 50 minutes but it never restarted and I don't see any unhealthy events in Health Events section in service fabric explorer.

<img width="910" alt="unhealthy_container" src="https://user-images.githubusercontent.com/36767136/37449576-cd2d94fe-2850-11e8-895e-92ea5e198a98.PNG">

<img width="754" alt="service_fabric_health_events" src="https://user-images.githubusercontent.com/36767136/37449439-3683f6f6-2850-11e8-97d7-d0b0f557687c.PNG">

Following is my setup

1. I have following HEALTHCHECK line in my dockerfile

"Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "powershell -command try {$response = iwr http://localhost/status/ping -UseBasicParsing; if ($response.StatusCode -eq 200) { return 0} else {return 1}; } catch { return 1 }"
                ],
                "Interval": 90000000000,
                "Timeout": 10000000000,
                "Retries": 3
            }

2. I have added following in ApplicationManifest ContainerHostPolicies section

HealthConfig IncludeDockerHealthStatusInSystemHealthReport="true" RestartContainerOnUnhealthyDockerHealthStatus="true"/

3. I am working on Service fabric standalone version "6.1.472.9494" and I have a one node cluster.

4. Following is the Hosting section from cluster configuration.

    Section Name="Hosting"
      Parameter Name="ContainerImageDownloadTimeout" Value="2400"
      Parameter Name="EnableDockerHealthCheckIntegration" Value="true" 
      Parameter Name="EndpointProviderEnabled" Value="true"
      Parameter Name="FirewallPolicyEnabled" Value="true"
      Parameter Name="RunAsPolicyEnabled" Value="true"
    Section

@mani-ramaswamy  Can you please provide some inputs on way forward to get this work.

Thanks,
Aditya Kanthe

**Comment by harahma** [comment]
@adityapkanthe Could you please share the Service Fabric traces from your one node cluster?

**Comment by adityapkanthe** [comment]
@harahma I was troubleshooting this issue further,

Firstly I was using AWS instance "Windows 2016 with containers" which included pre installed docker with following version.

 Client:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64

And healthcheck integration did not work with this docker version.

After that I created a clean AWS instance without any pre installed docker and then I manually installed following docker version.

Client:
 Version:      17.06.2-ee-6
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   e75fdb8
 Built:        Mon Nov 27 22:46:09 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.2-ee-6
 API version:  1.30 (minimum version 1.24)
 Go version:   go1.8.3
 Git commit:   e75fdb8
 Built:        Mon Nov 27 22:55:16 2017
 OS/Arch:      windows/amd64
 Experimental: false

The health check integration worked with this docker version.

May be supported docker version should be mentioned in documentation for health check integration.

Thanks,
Adiya Kanthe
